presto-spi/src/main/java/com/facebook/presto/spi/analyzer/AccessControlReferences.java

```diff
@@ -28,9 +28,11 @@ public class AccessControlReferences
 {
     private final Map<AccessControlRole, Set<AccessControlInfoForTable>> tableReferences;
     private final Map<AccessControlInfo, Map<QualifiedObjectName, Set<Subfield>>> tableColumnAndSubfieldReferencesForAccessControl;
+    private final AccessControlInfo queryAccessControlInfo;
```

Comment:
maybe make this non-final/settable. This class is already not threadsafe, the maps all get populated after this is constructed.  The advantage is then we get set this field in StatementAnalyzer when we call analyze(), instead of having to pass it through to analysis.  That should make the scope of changes much smaller (wouldn't need to make changes for everything that constructs an analysis object)